src-tauri/src/producer.rs

```diff
@@ -93,25 +109,38 @@ fn insert_new_items(db: &Connection, feed: i32, items: &[RawItem]) -> Vec<ItemTo
     inserted
 }
 
-fn get_most_recent_items(db: &Connection) -> HashMap<i32, DateTime<FixedOffset>> {
-    let opt = ItemReadOption {
-        ids: None,
-        feed: None,
-        status: None,
-        is_saved: None,
-        order_by: Some(ItemOrder::PublishedDateDesc),
-        limit: Some(1),
-        offset: None,
-    };
-
-    let rows = items::read_all(db, &opt).unwrap();
-
+fn get_most_recent_items(db: &Connection) -> Result<HashMap<i32, DateTime<FixedOffset>>> {
     let mut most_recent_items = HashMap::new();
-    for row in rows {
-        let feed = row.feed.id;
-        let published_at = row.published_at;
-        most_recent_items.insert(feed, published_at);
+
+    let feed_ids = get_all_feed_ids(db)?;
+
+    for feed_id in feed_ids {
+        let opt = ItemReadOption {
+            ids: None,
+            feed: Some(feed_id),
+            status: None,
+            is_saved: None,
+            order_by: Some(ItemOrder::PublishedDateDesc),
+            limit: Some(1),
+            offset: None,
+        };
+
+        match items::read_all(db, &opt) {
+            Ok(items) => {
+                if let Some(item) = items.first() {
+                    most_recent_items.insert(item.feed.id, item.published_at);
+                }
+            }
+            Err(err) => return Err(Error::InvalidValue(err.to_string())),
+        }
```

Comment:
I think it's better to use shortcuts for error handling. `items::read_all` will propagate the appropriate database error.

```suggestion
        if let Some(item) = items::read_all(db, &opt)?.first() {
            most_recent_items.insert(item.feed.id, item.published_at);
        }
```

Reply:
I agree with your suggestion. It's cool.